pkg/pool-manager/virtualmachine_pool.go

```diff
@@ -466,7 +467,11 @@ func (p *PoolManager) MarkVMAsReady(vm *kubevirt.VirtualMachine, parentLogger lo
 // mutating webhook but we didn't get the creation event in the controller loop
 // this mean the create was failed by some other mutating or validating webhook
 // so we release the virtual machine
-func (p *PoolManager) vmWaitingCleanupLook(waitTime int) {
+func (p *PoolManager) vmWaitingCleanupLook(waitTime int, mgr manager.Manager) {
+	if mgr != nil {
```

Comment:
Shouldn't we check if its the leader pod inside the mutex lock?

Reply:
I don't think we want to hold the mutex locked until the pod is elected. if it's our intention, it should be in a separate function that will document it properly.